### PR TITLE
JBPM-5296 - Cleaning up human-task and removing deprecated code

### PR DESCRIPTION
**Pull request by MarianMacik**
Changes for [JBPM-5296](https://issues.jboss.org/browse/JBPM-5296) within kie-api.

PR set

#199
droolsjbpm/jbpm#725

### CODE REVIEW

File: kie-api/src/build/revapi-config.json [hunk 1]
```diff
@@ -307,6 +307,16 @@
        "code": "java.method.addedToInterface",
        "new": "method <T> T org.kie.api.runtime.KieSession::getKieRuntime(java.lang.Class<T>)",
        "justification": "Promoting internal method getKieRuntime() to the public API"
+      },
+      {
+        "code": "java.method.removed",
+        "old": "method java.util.List<org.kie.api.task.model.TaskSummary> org.kie.api.task.TaskService::getTasksByVariousFields(java.lang.String, java.util.List<java.lang.Long>, java.util.List<java.lang.Long>, java.util.List<java.lang.Long>, java.util.List<java.lang.String>, java.util.List<java.lang.String>, java.util.List<java.lang.String>, java.util.List<org.kie.api.task.model.Status>, java.util.List<java.lang.String>, boolean)",
+        "justification": "Deprecated"
```
Comment (triceo) [inline kie-api/src/build/revapi-config.json]: Let's change this to something like "Removing API previously deprecated." Sentences generally look and work better for communicating things.

Comment (mswiderski) [inline kie-api/src/build/revapi-config.json]: +1

@MarianMacik could you please apply the suggested change?

**Comment by MarianMacik** [comment]
Change proposed by @triceo has been incorporated.

**Comment by triceo** [comment]
OK with me.

**Comment by MarianMacik** [comment]
jenkins retest this

**Pull request merged**